app/controllers/TransferAgreementComplianceController.scala

```diff
@@ -21,28 +22,52 @@ class TransferAgreementComplianceController @Inject() (
     val transferAgreementService: TransferAgreementService,
     val keycloakConfiguration: KeycloakConfiguration,
     val consignmentService: ConsignmentService,
-    val consignmentStatusService: ConsignmentStatusService
+    val consignmentStatusService: ConsignmentStatusService,
+    val configuration: Configuration
 )(implicit val ec: ExecutionContext)
     extends TokenSecurity
     with I18nSupport {
-  val transferAgreementForm: Form[TransferAgreementComplianceData] = Form(
+  val transferAgreementFormWithOpenRecords: Form[TransferAgreementComplianceData] = Form(
     mapping(
       "droAppraisalSelection" -> boolean
         .verifying("Departmental Records Officer (DRO) must have signed off the appraisal and selection decision for records", b => b),
       "droSensitivity" -> boolean
         .verifying("Departmental Records Officer (DRO) must have signed off sensitivity review", b => b),
       "openRecords" -> boolean
         .verifying("All records must be open", b => b)
-    )(TransferAgreementComplianceData.apply)(TransferAgreementComplianceData.unapply)
+    )((droAppraisalSelection, droSensitivity, openRecords) => TransferAgreementComplianceData(droAppraisalSelection, droSensitivity, Option(openRecords)))(data =>
+      Option(data.droAppraisalSelection, data.droSensitivity, data.openRecords.getOrElse(false))
+    )
   )
 
-  val taFormNamesAndLabels = Seq(
+  val transferAgreementForm: Form[TransferAgreementComplianceData] = Form(
+    mapping(
+      "droAppraisalSelection" -> boolean
+        .verifying("Departmental Records Officer (DRO) must have signed off the appraisal and selection decision for records", b => b),
+      "droSensitivity" -> boolean
+        .verifying("Departmental Records Officer (DRO) must have signed off sensitivity review", b => b)
+    )((droAppraisalSelection, droSensitivity) => TransferAgreementComplianceData(droAppraisalSelection, droSensitivity, None))(data =>
+      Option(data.droSensitivity, data.droAppraisalSelection)
+    )
+  )
+
+  private def form: Form[TransferAgreementComplianceData] =
+    if (blockClosureMetadata && blockDescriptiveMetadata) {
+      transferAgreementFormWithOpenRecords
+    } else {
+      transferAgreementForm
+    }
+
+  val blockClosureMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.closureMetadata")
+  val blockDescriptiveMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.descriptiveMetadata")
```

Comment:
It's just a suggestion. Feel free to ignore it.

Rename `FrontEndInfoConfiguration` to `ApplicationConfig` and get these config values from this class:

```

class ApplicationConfig @Inject()(configuration: Configuration) {

  private def get(location: String) = configuration.get[String](location)

  def frontEndInfo: FrontEndInfo = FrontEndInfo(
    get("consignmentapi.url"),
    get("environment"),
    get("region"),
    get("upload.url")
  )

  val blockClosureMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.closureMetadata")
  val blockDescriptiveMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.descriptiveMetadata")
}

```
Now, we can use these variables on the controllers. Actually, I don't want to hardcode the same string everywhere.

Reply:
Yep, good point. I've moved these out the config, updated the controllers and updated the tests. I've also updated the FileCheckResults controller which is also using these feature block variables.